apis/core/lib/domain/table/create.ts

```diff
@@ -0,0 +1,45 @@
+import { GraphPointer } from 'clownface'
+import { schema, rdf } from '@tpluscode/rdf-ns-builders'
+import { cc } from '@cube-creator/core/namespace'
+import { ResourceStore } from '../../ResourceStore'
+import * as id from '../identifiers'
+import { resourceStore } from '../resources'
+import { NamedNode } from 'rdf-js'
+
+interface CreateTableCommand {
+  tableCollection: GraphPointer
+  resource: GraphPointer
+  store?: ResourceStore
+}
+
+export async function createTable({
+  tableCollection,
+  resource,
+  store = resourceStore(),
+}: CreateTableCommand): Promise<GraphPointer> {
+  const label = resource.out(schema.name)
+  if (!label?.term) {
+    throw new Error('schema:name missing from the payload')
+  }
+
+  const csvMapping = tableCollection.out(cc.csvMapping)
+  if (!csvMapping?.term) {
+    throw new Error('cc:csvMapping missing from the payload')
+  }
+
+  if (!tableCollection?.term) {
```

Comment:
This is unnecessary because `tableCollection` will not be falsy

Reply:
The compiler complains that it can be undefined

Reply:
You sure? Do check again